zoom_sesh.py

```diff
@@ -6,73 +6,97 @@
   df = pd.read_csv(os.path.join(dir, name_files[0]))
   return df
 
-def make_fake_data(max_people=40):
-  df = import_names(NAMES_DIR)
+def make_fake_data(dir_name, max_people=40, overwrite=True):
+  df = import_random_names(NAMES_DIR)
   track_names = ['optics', 'semi', 'polymer', 'sensors']
   years = list(map(str, range(2013, 2020)))
   df.columns = ['name', 'year', 'track']
   df['track'] = [random.choice(track_names) for _ in range(len(df))]
   df['year'] = [random.choice(years) for _ in range(len(df))]
 
-  df = df.iloc[:max_people]  
   person_id = list(map(str,np.arange(max_people).tolist()))
 
-  for i in person_id:
-    df[i] = np.zeros(len(df))
-    df[i+"_cnsctv"] = np.zeros(len(df))
+  #df = ZoomSesh._create_tracking_cols(df.iloc[:max_people])
+  if os.path.isdir(dir_name):
+    if overwrite:
+      shutil.rmtree(dir_name)
+      os.mkdir(dir_name)
+    else:
+      raise ValueError(f'{dir_name} already exists! Please set `overwrite` to `False`')
+  else:
+    os.mkdir(dir_name)
+
+  w = pd.ExcelWriter(f'{dir_name}/alumni.xlsx')
+  df.iloc[:max_people].to_excel(w, index=False)
+  w.save()
+  w.close()
 
-  return df.iloc[:max_people]
 
 class ZoomSesh:
   '''Object that helps organize large groups of people during a zoom call.'''
 
-  def __init__(self, filename=None, max_people=40):
-    '''Constructor
+  def __init__(self, session_directory):
+    '''Creates a ZoomSesh from student file contained in session_directory
     Args:
-      filename: location of a file containing the alumni for this session. Columns
-        TBD. If `None`, then ZoomSesh will initialize with fake data of len `max_people`.
-  
-      max_people: int specifying number of people for fake data. Does not do anything
-        unless `filename` is None.
-  
+      session_directory: folder containing important information about a session. This folder
+        *must* contain a file called 'alumni.xlsx' containing the columns 'name', 'year',
+        and 'track'.
     '''
-    if filename is not None:
-      raise NotImplementedError('This feature is not ready yet')
-    
-    else:
-      # For development, testing, debugging, etc.
-      self._alumni_history = [make_fake_data(max_people=max_people)]
+
+    if 'alumni.xlsx' not in os.listdir(session_directory):
+      raise FileNotFoundError(f"No alumni file found! Please make sure to include 'alumni.xlsx' in {session_directory}")
+      
+    alumni_file = os.path.join(session_directory, ALUMNI_FILE)
+    self._session_directory = session_directory
+    self._alumni_data = pd.read_excel(alumni_file) # DataFrame with raw data from alumni file
+
+    # Attributes for this zoom session. These are taken from the column names of the alumni file
+    # For now, these will likely just be 'name', 'year' and 'track'
+    self.attributes = [col for col in self._alumni_data.columns]
+
+    self._alumni_data = self._create_tracking_cols((self._alumni_data))
```

Comment:
Yep, I now understand the tracking column creation

Reply:
Ah! Didn't scroll down before I commented above!